### ISSUE TITLE
Mission assumes that waypoint has been visited

### ISSUE THREAD

**Issue by Tadrix** [endless-sky/endless-sky#2235]
Related: #2179 

Mission: Stranded Freighter
Still being on Longjump, Sabik, reloaded the game to a point where already received the mission from Tomek, but didn't take off yet. On load got a message: "You have reached Longjump, but you left the F.S. Shackleton behind! Better depart and wait for them to arrive in this star system."
They cannot arrive since I wasn't there and didn't repair their ship.

**Comment by KiLEdEnNis** [comment]
when ya load game ya "reland", triggering missions with the "[on land?]" attribute, or basically triggering conversations in general
this waypoint thing happens with any escort mission if not all ships you are escorting & need to get to arrive are in the system
and the FW mission where ya loot a surveillance pod but already have 1 in cargo: reloading the save makes you continue without taking off (if I remember)
bla bla bla
in short: happens w/ basically any escort mission & has been around for as long as I remember, which makes me wonder why it isn't fixed yet

and about the Shackleton
I remember landing on Deep (Kornephoros) when I saved it to refuel, and when I departed it was disabled again

**Comment by Tadrix** [comment]
Another bug then?

Speaking of "re-landing" on load, I figured so much myself. Still, knowing what is happening changes little — the game is acting in a way user wouldn't reasonably expect. Maybe there is a way to change it?

I know that conversations that happen on landing happen on load too — I think we want to keep that, since save was made before conversation ended. We just need for the game to not treat as landings those saves which are automatically made on taking off.

**Comment by KiLEdEnNis** [comment]
loading reloads stuff like the derelict status & those mission messages, there's no easy way around, unless it's being changed when to reload stuff

**Comment by LocalGod79** [comment]
this is not a "new bug" - any time you visit a mission destination without completing the requirements, a dialog is displayed.

the dialog is not keyed to "how complete" the mission is - if you haven't finished *all* the requirements, the dialog will be displayed.

**Comment by KiLEdEnNis** [comment]
@LocalGod79 I think ya got confused a little (& not even by me o:)
he meant ya get the mission, then save/load, you get that message while ya never left the planet yet

**Comment by LocalGod79** [comment]
not confused - the issue is that when you load the character who's on a destination planet where there's an incomplete mission, "enter ship" involves "landing" there, and the incomplete mission dialog pops up...

**Comment by KiLEdEnNis** [comment]
so reloading game counts as relanding too? explains everything

**Comment by Bladewood** [comment]
Tadrix is opening this issue because it is weird to him.
I find it weird too.
We already know very well why and how the code makes it happens. **That is not the problem.**

**Comment by endless-sky** [comment]
Fixed in 686c035, although I think the broader solution will be to not show "on visit" messages when a game is first loaded.
